tests/test_type_checking_imports_transformer.py

```diff
@@ -0,0 +1,162 @@
+from typing import Type, List
+
+from libcst.codemod import CodemodTest, Codemod, CodemodContext
+from libcst.codemod.visitors import ImportItem
+
+from monkeytype.type_checking_imports_transformer import (
+    MoveImportsToTypeCheckingBlockVisitor,
+)
+
+
+class TestMoveImportsToTypeCheckingBlockVisitor(CodemodTest):
+
+    TRANSFORM: Type[Codemod] = MoveImportsToTypeCheckingBlockVisitor
+
+    def run_test_case(
+        self,
+        import_items_to_be_moved: List[ImportItem],
+        before: str,
+        after: str,
+    ) -> None:
+        context = CodemodContext()
+        MoveImportsToTypeCheckingBlockVisitor.store_imports_in_context(
+            context,
+            import_items_to_be_moved,
+        )
+        self.assertCodemod(before, after, context_override=context)
+
+    def test_simple_add_type_checking(self):
+        source = """
+            from __future__ import annotations
+            
+            from a import B
+            import c.C
+        """
+        import_items_to_be_moved = [
+            ImportItem('a', 'B'),
+            ImportItem('c.C'),
+        ]
+        expected = """
+            from __future__ import annotations
+            from typing import TYPE_CHECKING
+            
+            if TYPE_CHECKING:
+                import c.C
+                from a import B
+        """
+
+        self.run_test_case(import_items_to_be_moved, source, expected)
+
+    def test_type_checking_block_already_exists(self):
+        source = """
+            from __future__ import annotations
+            
+            from typing import TYPE_CHECKING
+
+            from a import B
+            import c.C
+            
+            if TYPE_CHECKING:
+                from d import E
+        """
+        import_items_to_be_moved = [
+            ImportItem('a', 'B'),
+            ImportItem('c.C'),
+        ]
+        expected = """
+            from __future__ import annotations
+            
+            from typing import TYPE_CHECKING
+            
+            if TYPE_CHECKING:
+                import c.C
+                from a import B
+            
+            if TYPE_CHECKING:
+                from d import E
```

Comment:
It would be nice if in this case we could find and append to the existing `if TYPE_CHECKING:` block rather than adding a new one. It seems like in principle that should be doable, but I'm assuming you looked at it and felt it was too complex? It could also always be done as a follow-up improvement if anyone cares enough.

Reply:
Correct, it should be do-able but it was getting too complicated, that's why took the debt, will try to fix this in the subsequent PR :)